streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1692,7 +1692,15 @@ class ZStream[-R, +E, +A] private[stream] (private[stream] val structure: ZStrea
         done <- Ref.make(false).toManaged_
         pull = done.get flatMap {
           if (_) Pull.end
-          else as.raceAttempt(p.await.mapError(Some(_)).ensuring(done set true) *> Pull.end)
+          else
+            as.raceAttempt(
```

Comment:
Hmm, should you set `done` to `true` if `as` completes successfully?

Reply:
Ah, I misread the bracketing. Looks great!

Reply:
Yeah we only toggle done if the interrupting promise resolves. Thanks for looking!